app/api/chat/engine/chat.ts

```diff
@@ -21,7 +21,16 @@ export async function createChatEngine({
     );
   }
 
-  const index = await getDataSource(datasource);
+  const isCollectionExist = await getMilvusClient().hasCollection({
+    collection_name: datasource,
+  });
+  if (!isCollectionExist.value) {
+    throw new Error(
+      `Collection "${datasource}" does not exist! Run the generate script or try uploading a file.`,
+    );
+  }
```

Comment:
how about we move this check into `getIndex`?

Reply:
how about adding a `checkExists` parameter to `getIndex` then?

`getIndex({datasource, checkExists: false})`